Modules/QuestieQuest.lua

```diff
@@ -423,7 +424,7 @@ end
 
 function QuestieQuest:UpdateQuest(QuestId)
     local quest = QuestieDB:GetQuest(QuestId);
-    if quest and not Questie.db.char.complete[QuestId] then
+    if quest and (not Questie.db.char.complete[QuestId] or quest.Repeatable) then
```

Comment:
I prefer this solution as it saves one call to the database and won't mess with the `complete` state elsewhere.

Reply:
Yeah, just not sure if the `complete` state is ever really accurate for repeatable quests. If we opt for this change, after looking at the code again, I think we'd also need to add checks for `quest.Repeatable` in the `CalculateAvailableQuests()` and `AddFinisher()` methods where we check the `complete` state.

Reply:
I also had another look at the code to check the use of `Questie.db.char.complete` and now I prefer the other solution. Thing is currently we mark repeatable quests as complete even though on the next reset/reload they will be unmarked. The API function `GetQuestsCompleted` (and similar functions) always return `false` for repeatable quests.

So we are actually storing it wrong at this point.